context/app/static/js/components/entity-search/MetadataMenu/MetadataMenu.jsx

```diff
@@ -38,16 +39,34 @@ function NotebookMenuItem(props) {
   );
 }
 
-function MetadataMenu({ entityType }) {
+function MetadataMenu({ entityType, results }) {
   const lcPluralType = `${entityType.toLowerCase()}s`;
   const { selectedHits, createNotebook, closeMenu } = useMetadataMenu(lcPluralType);
   const menuID = 'metadata-menu';
 
   const { selectedRows } = useStore();
-  const errorMessage =
-    selectedRows.size > 10
-      ? `You have selected ${selectedRows.size} datasets. Workspaces currently only supports up to 10 datasets. Please unselect datasets.`
-      : null;
+
+  // array that contains all the datasets from current pagination where mapped_data_access_level === 'Protected'.
+  const filteredDataset = results?.hits?.items?.filter(
+    (item) => item?.fields?.mapped_data_access_level === 'Protected',
+  );
+
+  const containsProtectedDataset = filteredDataset?.some((item) => {
+    const filteredDatasetIds = item?.id;
+    // spreading out the selectedRows Set to convert to an array then checking to see if any contains mapped_data_access_level === 'Protected'.
+    return [...selectedRows].some((selectedRowId) => filteredDatasetIds.includes(selectedRowId));
+  });
+
+  let errorMessage;
+
+  if (selectedRows.size > 10) {
+    errorMessage = `You have selected ${selectedRows.size} datasets. Workspaces currently only supports up to 10 datasets. Please unselect datasets.`;
+  } else if (containsProtectedDataset) {
+    errorMessage =
+      'You have selected protected datasets. Workspaces currently only supports published public datasets. Selected protected datasets are shown below.';
+  } else {
+    errorMessage = null;
+  }
```

Comment:
The algorithm needs to check many to many for all the ids of selectedRows against all the ids of filteredDatasetIds (the array that contains ids of protected datasets). 

Since we are dealing with two arrays, this is a double for loop, an n * m operation, n being the number of selected Rows, and m being the number of rows that are displayed on the current pagination when the table renders. 

However, n (the number of rows that are selected) has an upper bound of 10, because as soon as we selected more than 10 items, the other error message will be displayed. Moreover, m (the number of rows that are returned from the results/current pagination) is also bound by 18, since only 18 shows up at a time. 

So in the end, in the worst case scenario, the big O notation comes out to 10 * 18 operations, which is constant time. I thought this might be helpful since a double for loop was being implemented and the operation looks expensive. 

I am also including a screenshot in the scenario where more than 10 protected datasets are selected. 

![image](https://github.com/hubmapconsortium/portal-ui/assets/45410262/d463eacb-7739-4b98-99b9-33d5c24f25c3)